### ISSUE TITLE
Add semantic markup to layouts

### ISSUE THREAD

**Issue by wesruv** [backdrop/backdrop-issues#1970]
Re: https://github.com/biolithic/backdroplayouts/issues/29

We definitely need to get a `<main>` tag back in, one on ever page.

Now that I'm looking into the `<section>` tag again, we might be better off with `<div>`s. A section should have a headline, and is part of the document outline, neither of which are a given in how these layouts might be used.

**Comment by jenlampton** [comment]
> We definitely need to get a <main> tag back in, one on every page.

In most cases we can swap out the div with the class `l-content` with a main tag instead, but there are some layouts where there isn't a clear main section (Geary, Rolph, Sutro). Should we make an assumption for those layouts, and allow people to theme/override if they are doing something other than we assumed?

**Comment by klonos** [comment]
I think that it'd make a lot more sense if there was a way to dynamically assign the `<main>` tag to whichever region holds the "Main page content" block. That would be a very sensible default and we could provide an easy way for people to override it if they had edge cases.

**Comment by jenlampton** [comment]
That would make sense to the end-user, but it comes at a very high price to the theme developer.

I'm worried dynamically generating HTML tags would make a mess out of our layout template files, and our theme system is hard enough to understand already. Replacing THML tags that people instantly recognize with complicated php variables that need to be contemplated is a step backwards and will increase the learning curve.

I'd say let's use main where we *know* what the main area will be, and use divs everywhere else. But, I could be convinced to make an assumption for other layouts (center column in 3 col layout? left-most column in 2/4 column layout?) if others think that's a better approach.

**Comment by klonos** [comment]
Fair enough @jenlampton

**Comment by wesruv** [comment]
@jenlampton the main tag can wrap secondary content, it's just supposed to avoid site-wide content (e.g. site header and footer).

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main
> This content should be unique to the document, excluding any content that is repeated across a set of documents such as sidebars, navigation links, copyright information, site logos, and search forms (unless the document's main function is as a search form).

http://html5doctor.com/the-main-element/
> The main content of the body of a document or application. The main content area consists of content that is directly related to or expands upon the central topic of a document or central functionality of an application.

I think we should make a best guess, and it can be overridden for people who want it somewhere else.

I feel like it'd be better to overshoot the scope of `<main>` rather than not grab enough regions. I think it'd be safe to wrap everything that isn't the `header` and `footer`. I'm seeing less and less side bar navs nowadays.

If it helps, `<main>` servers a similar purpose to the 'skip to content' or 'skip to main' anchor links and is the same under the covers as putting `aria-role="main"` on a div in most modern browsers.

https://www.w3.org/TR/wai-aria/roles#main
> The main role is a non-obtrusive alternative for "skip to main content" links, where the navigation option to go to the main content (or other landmarks) is provided by the user agent through a dialog or by assistive technologies.

**Comment by klonos** [comment]
If I may say, by the definitions, it seems to me that the "Main page content" *block* itself should have the `<main>` tag. There will be of course layouts and themes that may not have that block, but it seems that those will be edge cases.

**Comment by wesruv** [comment]
I'd still say it's safer to capture too much in `<main>` than too little. For example, the basis 'Get Started' homepage, which could be done using a one column, or multi-row multi-column layout, `<main>` should wrap everything that isn't the `<header>` and `<footer>`

![image](https://cloud.githubusercontent.com/assets/5607236/13161958/51abbb88-d66f-11e5-8479-5344b177ccf0.png)

I would also point to the MDN homepage:
![image](https://cloud.githubusercontent.com/assets/5607236/16091108/37610bdc-3300-11e6-9170-2cb58504a0b1.png)

**Comment by wesruv** [comment]
@klonos the function you're thinking of is more like `<article>` or `<section>`. It's completely acceptable, and even expected, to have tags like `<aside>` inside of the `<main>` (aside which is generally used for sidebars).

**Comment by klonos** [comment]
So, where are we on this one? Should we have `<main>` in the "Main page content" block or the "Content" region or perhaps have it wrapping many regions (all but header/footer, top/bottom)?

**Comment by jenlampton** [comment]
I like wrapping the main columns. I could go either way about including top and/or bottom, but I'm leaning towards excluding them because they also exist on all other layouts and will not be inside the main tag there. How about we use it for either the container or the row for Geary, Rolph and Sutro?

**Comment by klonos** [comment]
Yeah, excluding top/bottom and header/footer seems more appropriate I think. Lets go with that then.

Although, I have to say that reusable things go into the sidebars too. The only region/block with unique content is, well. Content.

**Comment by jenlampton** [comment]
> The only region/block with unique content is, well. Content.

and `Content` is most likely not going to exist on these pages. That is why I left out `<main>` in the first place :)

**Comment by quicksketch** [comment]
> I'd still say it's safer to capture too much in `<main>` than too little.

I agree with this too. Making our default markup err on the side of caution is better than excluding things that might be considered primary to the site's content.

**Comment by jenlampton** [comment]
> How about we use it for either the container or therow for Geary, Rolph and Sutro?

So are we going with this, then? And if so, which one?
